study1/week3/best-time-to-buy-and-sell-stock.java

```diff
@@ -0,0 +1,22 @@
+//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/
+
+class Solution {
+    public int maxProfit(int[] prices) {
+        //time complexity is O(n)
+        //space complexity is O(1)
+      
+        int buy = prices[0];
+        int profit = 0;
+
+        for (int i = 1; i < prices.length; i++) {
+            if (prices[i] < buy) {
+                buy = prices[i];
+            }
+            else {
```

Comment:
Oh, I like this else statement, most of the students (and also me) calculated the max unconditionally.